packages/git/src/browser/git-scm-provider.ts

```diff
@@ -413,7 +416,23 @@ export class GitAmendSupport implements ScmAmendSupport {
     }
 
     public async reset(commit: string): Promise<void> {
-        await this.git.exec(this.repository, ['reset', commit, '--soft']);
+        if (commit === 'HEAD~' && await this.isHeadInitialCommit()) {
```

Comment:
In VS Code they seem to be able to handle this without using the filesystem.
https://github.com/microsoft/vscode/blob/master/extensions/git/src/commands.ts#L1464

Reply:
> In VS Code they seem to be able to handle this without using the filesystem

@svenefftinge you are totally correct.  `git update-ref -d HEAD` does the trick perfectly.  Thanks for pointing that out.  It tests fine.  I have amended the commit accordingly.